### DISCUSSION TITLE
taro3.4x   vue3.2x 做小程序，如何使用Echarts？ 目前好像没有什么好的方案可以使用。

### DISCUSSION THREAD

**Discussion by wangrongding** [NervJS/taro#11633] [Q & A] [answered]
### 这个特性解决了什么问题？
taro3.4x   vue3.2x 做小程序使用echarts

### 这个 API 长什么样？
taro3.4x   vue3.2x 做小程序可以方便的使用echarts。

**Comment by wangrongding** [comment]
大家有好的方案吗？希望可以告知一下，谢谢🧡

**Comment by TheKonka** [comment]
https://docs.taro.zone/blog/2018-09-18-taro-1-0-0#%E6%94%AF%E6%8C%81%E5%BC%95%E7%94%A8%E5%B0%8F%E7%A8%8B%E5%BA%8F%E7%AB%AF%E7%AC%AC%E4%B8%89%E6%96%B9%E7%BB%84%E4%BB%B6%E5%BA%93

https://github.com/ecomfe/echarts-for-weixin

我用的这个

**Comment by wangrongding** [comment, marked as answer]
感谢~🧡

最后用原生混合实现了https://docs.taro.zone/docs/next/hybrid

**Reply by wangrongding** [reply]
但是这样不是很友好，希望有更好的方法。

**Reply by OXXD** [reply]
可以考虑下我基于官方 `echarts-for-weixin` 封装的 `npm` 包 [https://github.com/OXXD/echarts-for-weixin](https:/github.com/OXXD/echarts-for-weixin)，原理也是与原生混合一样，不过可以自己再基于此封装成 `Vue` 组件

主要优势就是直接 `npm` 安装管理依赖更加方便，并且支持按需引入只使用到的图表组件打包大小会少很多

安装也很简单
```bash
npm install echarts echarts-for-weixin
```

**Reply by brookyu** [reply]
github链接失效了。请问您了解怎么taro3 vue项目里整合cax动画库吗？

**Comment by CXS-21** [comment]
目前用的是这个https://github.com/Cecilxx/echarts-taro3-react

**Reply by GoogooFly** [reply]
请问这个有具体实现代码吗 我的react taro3 使用一直报错

**Comment by lloydzhou** [comment]
对于echarts封装组件可以看看这个项目　[https://github.com/lloydzhou/tsxecharts](https://github.com/lloydzhou/tsxecharts)　还有一个vue版本 [https://github.com/lloydzhou/vuecharts](https://github.com/lloydzhou/vuecharts)

两个版本都能将直接使用echarts的时候一整个巨大的json配置换成原生的react/vue组件来使用

![image](https://user-images.githubusercontent.com/1826685/223918344-1f599912-16bb-4830-ba14-3a168c4c4e07.png)
